### ISSUE TITLE
Profiler suddenly stopped working

### ISSUE THREAD

**Issue by onionka** [GoogleCloudPlatform/cloud-profiler-python#21]
Hello GCloud Team,

I have a problem with this library last month or so. It suddenly stopped working, nothing got changed in the console nor code. Could you please help me? We are using this in 3 environments (one of them is production), but we are unable to make it work properly.

**How to reproduce**
I don't have precise steps to reproduce this problem (because it happened all the sudden), so I cannot replicate this issue. 

**Logs**
```
E 2019-10-30T12:13:46.389820758Z Failed to collect and upload profile: <built-in function profile_cpu> returned NULL without setting an error 
E 2019-10-30T12:13:46.187933854Z Failed to set ITIMER_PROF:  
E 2019-10-30T12:13:17.687470023Z Failed to collect and upload profile: <built-in function profile_cpu> returned NULL without setting an error 
E 2019-10-30T12:13:17.495344141Z Failed to set ITIMER_PROF:  
E 2019-10-30T12:12:34.063318304Z Failed to collect and upload profile: <built-in function profile_cpu> returned NULL without setting an error 
E 2019-10-30T12:12:33.852451918Z Failed to set ITIMER_PROF:  
E 2019-10-30T12:12:05.084790490Z Failed to collect and upload profile: <built-in function profile_cpu> returned NULL without setting an error 
E 2019-10-30T12:12:04.843544227Z Failed to set ITIMER_PROF:  
```

**Comment by Cmiroslaf** [comment]
I noticed that it might be a problem with profiling multithreaded applications. A similar thing is doing also tracer - it doesn't work for some unknown reason too.

I forgot to add, the only thing that is being profiled and traced is the main thread.

**Comment by Cmiroslaf** [comment]
Currently, I found out that decreasing `period_ms` from 2000ms to default 10ms removed this issue. 

- We are using python 3.7
- `google-cloud-profiler = "==1.0.9"`
- `cpu time`
- it started right away, I mean there was not a single profile send, always that error
- I didn't know exactly how it started but now I remembered - it was because I used 2000ms in `period_ms` parameter
- We are using python3.7-alpine image:
```
FROM python:3.7-alpine AS base

RUN set -ex; \
    apk --no-cache add ca-certificates wget; \
    wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub; \
    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.29-r0/glibc-2.29-r0.apk; \
    apk add glibc-2.29-r0.apk; \
    apk update; \
    apk --no-cache add --update openssl ca-certificates bash gcc libxml2-dev libxslt-dev postgresql-dev python3-dev make libexecinfo-dev; \
    update-ca-certificates; \
    pip3 install pipenv

FROM base AS builder

COPY Pipfile /app/Pipfile
COPY Pipfile.lock /app/Pipfile.lock
WORKDIR /app

RUN apk --no-cache add --update build-base
RUN pipenv sync

FROM base

COPY --from=builder /root/.local/share/virtualenvs /root/.local/share/virtualenvs
```

But the problem with this is that 2s is that we have a lot microservices that are being profiled and 10ms is crossing the limit that I cannot move. I calculated that we need around 1000ms for it to not to cross that limit.

**Comment by nolanmar511** [comment]
What limit are you referring to? (In "2s was set up because 10ms is always crossing the limit that I cannot change. I calculated that we need around 1000ms for it to not cross that limit.").

**Comment by Cmiroslaf** [comment]
I meant quota in GCloud:

> Failed to collect and upload profile: <HttpError 429 when requesting https://cloudprofiler.googleapis.com/v2/projects/dashlyqa/profiles/xxx?alt=json returned "Quota exceeded for quota metric 'cloudprofiler.googleapis.com/update_requests' and limit 'UpdatesPerMinutePerProject' of service 'cloudprofiler.googleapis.com' for consumer 'project_number:xxx'.". Details: "[{'@type': 'type.googleapis.com/google.rpc.Help', 'links': [{'description': 'Google developer console API key', 'url': 'https://console.developers.google.com/project/xxx/apiui/credential'}]}]">

Ok then, so I will try it with python on Ubuntu and try the 2 seconds in `period_ms` parameter as I mentioned before.

**Comment by nolanmar511** [comment]
Okay, thanks! 

Changing the sampling rate won't impact the number of UpdateRequests per minute. It just means that each uploaded profile will have less samples in it (or, at least if it were totally WAI). 

Feel free to send a request to increase quota update_requests per minute -- it's often necessary when profiling a lot of microservices. https://cloud.google.com/profiler/quotas explains more about Stackdriver Profiler's quotas.

**Comment by jqll** [comment]
Hi, Cmiroslaf@, thanks for reporting! You are right that the issue is related to setting period_ms to 2s. There is a bug in the code that prevents using sample interval greater than 1s. We'll work on a fix. 

But as nolanmar511@ mentioned, the sampling rate won't impact the number of UpdateProfile requests. You typically don't want to customize the sample interval. It means how frequently we sample the stacks. For CPU profiler, we profile for 10s. So 2s sample interval means we sample the stack 5 times. This won't generate very interesting profile.

**Comment by kalyanac** [comment]
> As a side note: we do not (and have not previously) tested on alpine.
> While we do plan to add support for alpine, alpine is not supported now.

Alpine is now supported. See [Profiling Python applications  |  Cloud Profiler  |  Google Cloud](http://cloud/profiler/docs/profiling-python#running_with_linux_alpine)

**Comment by jqll** [comment]
Fixed by 0508aaf677fe704f6c1671c0f56400949c0150e2.
